src/dotnet/ShellShim/WindowsEnvironmentPath.cs

```diff
@@ -31,17 +30,27 @@ public void AddPackageExecutablePathToUserPath()
                 return;
             }
 
-            var existingUserEnvPath = Environment.GetEnvironmentVariable(PathName, EnvironmentVariableTarget.User);
+            var existingUserEnvPath = _environmentProvider.GetEnvironmentVariable(PathName, EnvironmentVariableTarget.User);
 
             if (existingUserEnvPath.EndsWith(';'))
             {
-                existingUserEnvPath = existingUserEnvPath.Substring(0, (existingUserEnvPath.Length - 1));
+                existingUserEnvPath = existingUserEnvPath.TrimEnd(';');
```

Comment:
@peterhuene nit: i am aware of TrimEnd, but what happen if the user actually have ";;;;".

Reply:
They get a PATH set to `foo;bar;<tool_path>` instead of `foo;bar;;;;<tool_path>`? Can revert if it makes a difference?

Reply:
the existing PATH is "foo;bar;;" already. Depends on you, I feel it is "too smart" to correct this. But i don't have strong opinion. Both are fine.

Reply:
I'll revert. If the user had it in their PATH to begin with, who are we to remove it?